pkgs/development/python-modules/plotly/default.nix

```diff
@@ -75,54 +74,22 @@ buildPythonPackage rec {
-    "test_orca_executable_path"
-    # FAILED plotly/tests/test_orca/test_orca_server.py::test_orca_version_number - ValueError:
-    "test_orca_version_number"
-    # FAILED plotly/tests/test_orca/test_orca_server.py::test_ensure_orca_ping_and_proc - ValueError:
-    "test_ensure_orca_ping_and_proc"
-    # FAILED plotly/tests/test_orca/test_orca_server.py::test_server_timeout_shutdown - ValueError:
-    "test_server_timeout_shutdown"
-    # FAILED plotly/tests/test_orca/test_orca_server.py::test_external_server_url - ValueError:
-    "test_external_server_url"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_simple_to_image[eps] - ValueError:
-    "test_simple_to_image"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_to_image_default[eps] - ValueError:
-    "test_to_image_default"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_write_image_string[eps] - ValueError:
-    "test_write_image_string"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_write_image_writeable[eps] - ValueError:
-    "test_write_image_writeable"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_write_image_string_format_inference[eps] - ValueError:
-    "test_write_image_string_format_inference"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_write_image_string_bad_extension_failure - assert 'must be specified as one of the followi...
-    "test_write_image_string_bad_extension_failure"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_write_image_string_bad_extension_override - ValueError:
-    "test_write_image_string_bad_extension_override"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_topojson_fig_to_image[eps] - ValueError:
-    "test_topojson_fig_to_image"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_latex_fig_to_image[eps] - ValueError:
-    "test_latex_fig_to_image"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_problematic_environment_variables[eps] - ValueError:
-    "test_problematic_environment_variables"
-    # FAILED plotly/tests/test_orca/test_to_image.py::test_invalid_figure_json - assert 'Invalid' in "\nThe orca executable is required in order to e...
-    "test_invalid_figure_json"
-    # FAILED test_init/test_dependencies_not_imported.py::test_dependencies_not_imported - AssertionError: assert 'plotly' not in {'IPython': <module>
-    "test_dependencies_not_imported"
-    # FAILED test_init/test_lazy_imports.py::test_lazy_imports - AssertionError: assert 'plotly' not in {'IPython': <module 'IPython' from '...
-    "test_lazy_imports"
     # requires vaex and polars, vaex is not packaged
     "test_build_df_from_vaex_and_polars"
     "test_build_df_with_hover_data_from_vaex_and_polars"
+    # lazy loading error, could it be the sandbox PYTHONPATH?
+    # AssertionError: assert "plotly" not in sys.modules
+    "test_dependencies_not_imported"
+    "test_lazy_imports"
```

Comment:
These fail because the correct libraries aren't available at check time. Put only the `pytestCheckHook` in `nativeCheckInputs` and the rest (save for optional dependencies) into `checkInputs`. That will fix these tests.

From the `stdenv` documentation:

> Dependencies needed only to run tests are similarly classified between native (executed during build) and non-native (executed at runtime):
> `nativeCheckInputs` for test tools needed on $PATH (such as ctest) and [setup hooks](https://ryantm.github.io/nixpkgs/stdenv/stdenv/#ssec-setup-hooks) (for example [pytestCheckHook](https://ryantm.github.io/nixpkgs/languages-frameworks/python/#python))
> `checkInputs` for libraries linked into test executables (for example the qcheck OCaml package)
> These dependencies are only injected when [doCheck](https://ryantm.github.io/nixpkgs/stdenv/stdenv/#var-stdenv-doCheck) is set to true.

Reply:
tried it, doesn't fix the lazy loading tests. python checkInputs are loaded at build time and will thus break cross

Reply:
How strange. Worked for me, even on cross. But again, you have more history with this so as you will.